cclib/parser/turbomoleparser.py

```diff
@@ -536,5 +536,127 @@ def after_parsing(self):
                     i -= 1
                 i += 1
 
-#EOF
+
+# Implementation for the new code begins here.
+class Turbomole(logfileparser.Logfile):
+    """A Turbomole log file"""
+
+    def __init__(self, *args, **kwargs):
+
+        # Call the __init__ method of the superclass
+        super(Turbomole, self).__init__(logname="Turbomole", *args, **kwargs)
+
+    def __str__(self):
+        """Return a string representation of the object."""
+        return "Turbomole output file %s" % (self.filename)
+
+    def __repr__(self):
+        """Return a representation of the object."""
+        return 'Turbomole("%s")' % (self.filename)
+
+    # This still needs to be implemented.
+    def normalisesym(self, label):
+        """Normalise the symmetries used by Turbomole."""
+
+    def before_parsing(self):
+        self.geoopt = False # Is this a GeoOpt? Needed for SCF targets/values.
+
+    def extract(self, inputfile, line):
+        """Extract information from the file object inputfile."""
+
+        ## This information is in the control file.
+        #   $rundimensions
+        #   dim(fock,dens)=1860
+        #   natoms=20
+        #   nshell=40
+        #   nbf(CAO)=60
+        #   nbf(AO)=60
+        #   dim(trafo[SAO<-->AO/CAO])=60
+        #   rhfshells=1
+        if line[3:10]=="natoms=":
+            self.natom=int(line[10:])
+
+        if line[3:11] == "nbf(AO)=":
+            nmo = int(line.split('=')[1])
+            self.set_attribute('nbasis', nmo)
+            self.set_attribute('nmo', nmo)
+
+        ## Atomic coordinated in job.last:
+        #              +--------------------------------------------------+
+        #              | Atomic coordinate, charge and isotop information |
+        #              +--------------------------------------------------+
+        #
+        #
+        #              atomic coordinates              atom shells charge pseudo isotop
+        #    -2.69176330   -0.00007129   -0.44712612    c      3    6.000    0     0
+        #    -1.69851645   -0.00007332    2.06488947    c      3    6.000    0     0
+        #     0.92683848   -0.00007460    2.49592179    c      3    6.000    0     0
+        #     2.69176331   -0.00007127    0.44712612    c      3    6.000    0     0
+        #     1.69851645   -0.00007331   -2.06488947    c      3    6.000    0     0
+        #...
+        #    -7.04373606    0.00092244    2.74543891    h      1    1.000    0     0
+        #    -9.36352819    0.00017229    0.07445322    h      1    1.000    0     0
+        #    -0.92683849   -0.00007461   -2.49592179    c      3    6.000    0     0
+        #    -1.65164853   -0.00009927   -4.45456858    h      1    1.000    0     0
+        if 'Atomic coordinate, charge and isotop information' in line:
+            self.skip_lines(inputfile,['border','b','b'])

```

Comment:
There should normally be [a space after commas](https://stackoverflow.com/questions/31741418/pep8-space-after-a-comma).